apiserver/facades/agent/uniter/uniter.go

```diff
@@ -3579,16 +3639,16 @@ func (u *UniterAPI) commitHookChangesForOneUnit(unitTag names.UnitTag, changes p
 			return errors.Trace(err)
 		}
 
-		// This API method never supported opening a port across multiple
-		// subnets. Instead, it was assumed that the port range was
-		// always opened in all subnets. To emulate this behavior, we
-		// simply open/close the requested port ranges for all endpoints.
 		for _, r := range changes.OpenPorts {
 			// Ensure the tag in the port open request matches the root unit name
 			if r.Tag != changes.Tag {
 				return apiservererrors.ErrPerm
 			}
-			unitPortRanges.Open("", corenetwork.PortRange{
+
+			// Pre-2.9 clients do not populate the new Endpoint
```

Comment:
Pre-2.9 is clear now, but later it might be take more time to track down that this is clients using facade version 16 or earlier.  Perhaps both 2.9 / v 16 or earlier would be good.

Reply:
Sure thing.